### PR TITLE
Add SimpleFormIterator label function

### PR DESCRIPTION
**Pull request by iamstiil**
Hopefully this finally works 😅 

Original PRs: #6292 #6294

### CODE REVIEW

**Review by djhi — CHANGES_REQUESTED** [review]

**Review by djhi — CHANGES_REQUESTED** [review]
Can you also update the documentation? https://github.com/marmelab/react-admin/blob/master/docs/Inputs.md#arrayinput

File: docs/Inputs.md [hunk 1]
```diff
@@ -1211,6 +1211,19 @@ import { ArrayInput, SimpleFormIterator, DateInput, TextInput } from 'react-admi
 </ArrayInput>
 ```
 
+Furthermore, if you want to manipulate the label displayed left of each item, you can pass a function to `SimpleFormIterator` via the `labelFn` prop.
```
Comment (djhi) [inline docs/Inputs.md:1214]:
```suggestion
Furthermore, if you want to customize the label displayed for each item, you can pass a function to `SimpleFormIterator` via the `labelFn` prop.
```

Depending on the theme, the label might not be on the left

**Review by djhi — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/form/SimpleFormIterator.tsx [hunk 2]
```diff
@@ -113,6 +114,7 @@ const SimpleFormIterator: FC<SimpleFormIteratorProps> = props => {
         margin,
         TransitionProps,
         defaultValue,
+        labelFn = DefaultLabelFn,
```
Comment (djhi) [inline packages/ra-ui-materialui/src/form/SimpleFormIterator.tsx:117]:
```suggestion
        getItemLabel = DefaultLabelFn,
```

File: packages/ra-ui-materialui/src/form/SimpleFormIterator.tsx [hunk 3]
```diff
@@ -197,7 +199,7 @@ const SimpleFormIterator: FC<SimpleFormIteratorProps> = props => {
                                 variant="body1"
                                 className={classes.index}
                             >
-                                {index + 1}
+                                {labelFn(index)}
```
Comment (djhi) [inline packages/ra-ui-materialui/src/form/SimpleFormIterator.tsx:202]:
```suggestion
                                {getItemLabel(index)}
```

File: packages/ra-ui-materialui/src/form/SimpleFormIterator.tsx [hunk 4]
```diff
@@ -326,6 +328,7 @@ export interface SimpleFormIteratorProps
     disabled?: boolean;
     disableAdd?: boolean;
     disableRemove?: boolean | DisableRemoveFunction;
+    labelFn?: (index: number) => string;
```
Comment (djhi) [inline packages/ra-ui-materialui/src/form/SimpleFormIterator.tsx:331]:
```suggestion
    getItemLabel?: (index: number) => string;
```

File: packages/ra-ui-materialui/src/form/SimpleFormIterator.spec.tsx [hunk 5]
```diff
@@ -499,6 +498,34 @@ describe('<SimpleFormIterator />', () => {
...
+                    <SideEffectContextProvider value={sideEffectValue}>
+                        <SimpleForm
+                            record={{
+                                id: 'whatever',
+                                emails: [{ email: 'foo' }, { email: 'bar' }],
+                            }}
+                        >
+                            <ArrayInput source="emails">
+                                <SimpleFormIterator
+                                    labelFn={index => `3.${index}`}
```
Comment (djhi) [inline packages/ra-ui-materialui/src/form/SimpleFormIterator.spec.tsx:514]:
```suggestion
                                    getItemLabel={index => `3.${index}`}
```

File: docs/Inputs.md [hunk 6]
```diff
@@ -1211,6 +1211,19 @@ import { ArrayInput, SimpleFormIterator, DateInput, TextInput } from 'react-admi
 </ArrayInput>
 ```
 
+Furthermore, if you want to customize the label displayed for each item, you can pass a function to `SimpleFormIterator` via the `labelFn` prop.
```
Comment (djhi) [inline docs/Inputs.md:1214]:
```suggestion
Furthermore, if you want to customize the label displayed for each item, you can pass a function to `SimpleFormIterator` via the `getItemLabel` prop.
```

File: docs/Inputs.md [hunk 7]
```diff
@@ -1211,6 +1211,19 @@ import { ArrayInput, SimpleFormIterator, DateInput, TextInput } from 'react-admi
...
+
+```jsx
+import { ArrayInput, SimpleFormIterator, DateInput, TextInput } from 'react-admin';
+
+<ArrayInput source="backlinks">
+    <SimpleFormIterator labelFn={(index) => `${index + 1}. link`}>
```
Comment (djhi) [inline docs/Inputs.md:1220]:
```suggestion
    <SimpleFormIterator getItemLabel={(index) => `${index + 1}. link`}>
```

**Review by djhi — CHANGES_REQUESTED** [review]

**Review by djhi — APPROVED** [review]

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**